### PR TITLE
Add kron(): kronecker product for 2D matrices

### PR DESCRIPTION
**Pull request by notmgsk**
Closes https://github.com/rust-ndarray/ndarray/issues/652

H/T @ethanhs

### CODE REVIEW

**Review by ethanhs — APPROVED** [review]
LGTM

**Review by bluss — APPROVED** [review]
Looks straightforward

**Review by bluss — CHANGES_REQUESTED** [review]
Oh, request changes would be the rather better summary. :slightly_smiling_face:

File: src/linalg/impl_linalg.rs [hunk 1]
```diff
@@ -691,6 +691,30 @@ unsafe fn general_mat_vec_mul_impl<A, S1, S2>(
     }
 }
 
+/// Kronecker product of 2D matrices.
+///
+/// The kronecker product of a LxN matrix A and a MxR matrix B is a (L*M)x(N*R)
+/// matrix K formed by the block multiplication A_ij * B.
+pub fn kron<T>(a: &Array2<T>, b: &Array2<T>) -> Array2<T>
```
[the remark this answers is not in the document]
Comment (ethanhs) [reply src/linalg/impl_linalg.rs:698]: I think `kron` would align with numpy, as a datapoint, just wanted to mention. It also is an operation usually done on 2 dimensional matrices, though I believe numpy supports higher dimensional kronecker products.

**Comment by ethanhs** [comment]
@notmgsk Are you interested in continuing to work on this? If not I will probably build off of this PR and respond to the review.

**Comment by ethanhs** [comment]
I went ahead and made #1105, which I think responds to the feedback here.

**Comment by bluss** [comment]
Superseded by  #1105

**Pull request closed without merging**